offer-manager-app/src/main/groovy/life/qbic/portal/offermanager/components/offer/create/CreateOfferViewModel.groovy

```diff
@@ -95,6 +96,23 @@ class CreateOfferViewModel {
         subscribeToResources()
     }
 
+    void addItem(ProductItemViewModel item) {
+        // we don't do anything when the amount is equal or smaller zero
+        if (item.quantity <= 0.0) {
+            return
+        }
+        List<ProductItemViewModel> alreadyExistingItems = productItems.findAll {
+            (ProductItemViewModel currentItem) ->
+                    currentItem.product.productId == item.product.productId
+        } as List<ProductItemViewModel>
+        double totalAmount = item.quantity
+        for (ProductItemViewModel currentItem : alreadyExistingItems) {
+            totalAmount = totalAmount + currentItem.quantity
```

Comment:
@jenniferboedker here your "error" originates. The quantity is added instead of being set.

Reply:
that is not correct. We want to add it to the already existing items. The question is, why the filter did not work for the product id.